### PR TITLE
bug where searching without ajax and using multi filter fails

### PR DESCRIPTION
**Pull request by djay**
and bug which prevented any robot tests running for 5.0
and a bug that broke the filter portlet when sorting on python 2.7

### CODE REVIEW

**Comment by djay** [comment]
@petschki there is another bug with sorting in python 2.7. 
```
        parsedquery = queryparser.parseFormquery(
            self.context, query, sort_on, sort_order)
```
in plone.app.querystring.querybuilder turns sort_order into unicode which then makes the search fail.
Any idea why?

**Comment by petschki** [comment]
> Any idea why?

no ... where should this conversion happen? I don't see it here 

https://github.com/plone/plone.app.querystring/blob/master/plone/app/querystring/queryparser.py#L20-L69

**Comment by djay** [comment]
> > Any idea why?
> 
> no ... where should this conversion happen? I don't see it here
> 
> https://github.com/plone/plone.app.querystring/blob/master/plone/app/querystring/queryparser.py#L20-L69

bad debugging. it actually happened earlier. The problem was safe_decode for the whole request. Seems like a bad idea but I've put a specific exception in

File: src/collective/collectionfilter/filteritems.py [hunk 1]
```diff
@@ -81,6 +81,9 @@ def get_filter_items(
 
     # Recursively transform all to unicode
     request_params = safe_decode(request_params)
+    # Things break if sort_order is not a str
+    if "sort_order" in request_params:
```
Comment (petschki) [inline src/collective/collectionfilter/filteritems.py:85]: I'd recommend to do this only in PY2 with:
```
if six.PY2 and "sort_order" in request_params:
```

Comment (djay) [reply src/collective/collectionfilter/filteritems.py]: ok. but its the same either way. in py3 its a str also.

**Comment by djay** [comment]
@petschki can you or the owner of this repo go in an change the coveralls settings to reduce the threshold? https://coveralls.io/github/collective/collective.collectionfilter/settings

File: src/collective/collectionfilter/filteritems.py (hunk 1 above)
Comment (petschki) [reply src/collective/collectionfilter/filteritems.py]: AFAIK in PY3 everything should be a str (which is unicode in PY2)

**Comment by petschki** [comment]
> @petschki can you or the owner of this repo go in an change the coveralls settings to reduce the threshold? https://coveralls.io/github/collective/collective.collectionfilter/settings

done

**Comment by djay** [comment]
@petschki tests are passing. (but with less coverage now due to putting in that condition). Can you merge?
btw. coverage was how I discovered that most of the tests weren't running for 5.0 in the first place.

**Review by petschki — APPROVED** [review]
LGTM

**Pull request merged**